modules/gapi/include/opencv2/gapi/gproto.hpp

```diff
@@ -154,6 +154,8 @@ bool GAPI_EXPORTS can_describe(const GMetaArg&  meta,  const GRunArgP& argp);
 void GAPI_EXPORTS validate_input_arg(const GRunArg& arg);
 void GAPI_EXPORTS validate_input_args(const GRunArgs& args);
 
+bool validate_input_meta_arg(const GMetaArg& meta, std::ostream* tracer = nullptr);
+bool validate_input_meta(const GMatDesc& meta, std::ostream* tracer = nullptr);
```

Comment:
Why do we need this here? It is a public header - it is exposed in the installed library package. Is it supposed to be used somehow by end users? I believe it is not

Reply:
it's actually the common part between gproto.cpp & gcompiled.cpp -which uses the same validation fucntionality

I've added it similarly in the same way with `validate_input_args` because i got the impression after file inspection that entire file exposes a set of auxiliary functions  - additionally, i didn't face with usage other function in this file as public interface. This overall assumption score told me that file is utility file
But maybe I wrong and this utility-like function should be implemented in other place

Reply:
moved to `src/ai/gmeta_chec.hpp/cpp`

Reply:
`gproto.cpp` and `gproto_priv.hpp` are the right ones.